File: 551 - Student Attendence Record I.java
class Solution {
    public boolean checkRecord(String s) {
        int absentCount = 0, lateContinuous = 0;
        char[] str = s.toCharArray();
        char last = '\0';
        for (char c : str){
            if (c == 'A') {
                absentCount++;
            }
            if (c == 'L'){
                if (last == 'L') {
                    lateContinuous++;
                } else {
                    lateContinuous = 1;
                }
            }
            if (absentCount > 1 || lateContinuous > 2) {
                break;
            }
            last = c;
        }
        return absentCount <= 1 && lateContinuous <= 2;

    }
}

File: 551 - Student Attendence Record I.py
class Solution:
    def checkRecord(self, s: str) -> bool:
        absent_count = 0
        late_continuous = 0
        last = ''
        for char in s:
            if char == 'A':
                absent_count += 1
            if char == 'L':
                if last == 'L':
                    late_continuous += 1
                else:
                    late_continuous = 1
            if absent_count > 1 or late_continuous > 2:
                break
            last = char
        return absent_count <= 1 and late_continuous <= 2


s = Solution()
print(s.checkRecord("LLLALL"))
print(s.checkRecord('PPALL'))
